### ISSUE TITLE
Invalid Version when using multiple esprima-fb versions

### ISSUE THREAD

**Issue by HenryTheHamster** [rstacruz/pnpm#380]
### pnpm version:
0.37.0

### Code to reproduce the issue:

Add:
```
"dependencies": {
    "browserify-shim": "^3.8.11",
    "envify": "^3.4.1"
 }
```
to package.json

### Expected behavior:
Installs packages

### Actual behavior:
Invalid Version: 3001.0001.0000-dev-harmony-fb

### Additional information:

Installing either library separately works fine - I suspect it's the combination of trying to pull two different versions of the esprima-fb library that causes the issue. 
I have seen suggested elsewhere (https://github.com/substack/astw/issues/6) that this is a known bad version, so there might not be anything PNPM can do here, though it should be noted that `npm i` works fine with that package.json (possibly related to https://github.com/npm/npm/issues/4978 regarding difference in id and url)

**Comment by zkochan** [comment]
I'll fix this today. This is because pnpm does not interpret semver versions and ranges loosely and npm does.

**Comment by zkochan** [comment]
Landed in [v0.38.0](https://github.com/rstacruz/pnpm/releases/tag/v0.38.0)
